drafts/sorting-data-with-clojure.md

```diff
@@ -0,0 +1,140 @@
+Sorting data with Clojure
+=========================
+
+It is unavoidable, really.
+
+Any data eventually needs to be sorted for presentation. Most of the times we’re very lucky and we could lean on the implicit order of data returned from the database, or we can decorate that HTML table with DataTables and get sorting for free.
+
+Implicit sorting is a crutch that I’ve relied on many times, and I’m sure you have too.
+
+## Implicit sorting in RDBMS systems
+
+What many of us observe when using, say MySQL, is that the rows get returned to us in an order we're familiar with. The insertion order. Using InnoDB tables this generally means ascending by numeric auto-incrementing primary keys. For MyISAM it is strictly insertion order. Other storage engines might have different properties.
+
+Already it is clear that within one RDBMS system we can have multiple behaviours depending on the underlying storage engine used by the table.
+
+Long story short, we have no gaurantees and this can easily nip us in the behind if we're not careful.
+
+## Leaving it to the caller
+
+Knowing now that we cannot depend on the source to sort the data for us, it is almost certainly up to the caller to sort the data. This often comes in the form of a query parameter, be it an `ORDER BY` clause in a SQL statement or a query parameter to an API.
+
+More bespoke sorting tends to happen in the consuming code, not at the source, and this is where we'll look at what Clojure offers us.
+
+# Sorting data with Clojure
+
+Clojure is great at taming data of all kinds, here I just want to explore a few ways to get some order to your data using simple functions.
+
+## Example data
+
+For the examples below I’m going to be working with some invoice data. Each invoice looks something like this:
+
+```clojure
+{:invoice/number “ACMEINV00001”
+ :invoice/date “2016-11-25”
+ :invoice/total-before-tax 100
+ :invoice/tax 10
+ :invoice/items [...]}
+```
+
+I’ll leave it up to you to imagine how rich these data structures can become in a real invoicing system.
+
+## Getting started with (sort-by)
+
+The first requirement could be to sort a vector of invoices by total. These is relatively simple with `(sort-by)`:
```

Comment:
I wouldn't refer to function names with parens around them -- `(sort-by)` -- unless you're showing a valid invocation. The naked `sort-by` is fine.